scalding-core/src/main/scala/com/twitter/scalding/serialization/Boxed.scala

```diff
@@ -0,0 +1,470 @@
+/*
+Copyright 2015 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package com.twitter.scalding.serialization
+
+import java.util.concurrent.atomic.AtomicReference
+import java.io.{ InputStream, OutputStream }
+
+import com.esotericsoftware.kryo.{ Serializer => KSerializer, DefaultSerializer, Kryo }
+import com.esotericsoftware.kryo.io.{ Input, Output }
+
+/**
+ * This interface is a way of wrapping a value in a marker class
+ * whose class identity is used to control which serialization we
+ * use. This is an internal implementation detail about how we
+ * interact with cascading and hadoop. Users should never care.
+ */
+trait Boxed[+K] {
+  def get: K
+}
+
+class BoxedDefaultSerialization extends KSerializer[Boxed[_]] {
+  def write(kryo: Kryo, output: Output, t: Boxed[_]) {
+    sys.error(s"Kryo should never be used to serialize a boxed instance: $t")
+  }
+  def read(kryo: Kryo, input: Input, t: Class[Boxed[_]]): Boxed[_] = {
+    sys.error("Kryo should never be used to serialize a boxed instance, class: $t")
+  }
+}
+
+// TODO: Make more of these with a script
+
+@DefaultSerializer(classOf[BoxedDefaultSerialization])
```

Comment:
I think we should remove this and instead register a default serializer for subclasses of Boxed. This way, this code might eventually get away with not depending on Kryo. Maybe not needed immediately.